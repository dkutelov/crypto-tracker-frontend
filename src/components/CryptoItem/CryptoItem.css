td {
  padding: 0.25em;
  font-size: 0.7em;
}

@media only screen and (min-width: 768px) {
  td {
    padding: 0.5em 1em;
    font-size: 1em;
  }
}

.coinImageWrapper {
  display: none;
}

@media only screen and (min-width: 768px) {
  .coinImageWrapper {
    display: inline-block;
    width: 40px;
    height: 40px;
  }
}

@media only screen and (max-width: 768px) {
  .noShowOnSmall {
    display: none;
  }
}

.coinImage {
  max-width: 100%;
}

.coinData {
  padding: 0.25;
}

@media only screen and (min-width: 768px) {
  .coinData {
    padding: 1em 2em;
  }
}

.coinPrice {
  text-align: right;
}

.priceDown {
  color: red;
}
.priceUp {
  color: lawngreen;
}
