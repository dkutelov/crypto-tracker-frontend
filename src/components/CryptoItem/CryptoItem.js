import { Link } from 'react-router-dom';

import './CryptoItem.css';
import { formatPrice } from '../../utils/formatPrice';
import MoreIcon from '../../assets/icons/more.svg';

const CryptoItem = ({
  id,
  name,
  image,
  symbol,
  current_price: price,
  high_24h: highestPrice,
  low_24h: lowestPrice,
  price_change_percentage_24h: priceChange,
  total_volume: volume,
}) => {
  return (
    <tr>
      <td className='coinImageWrapper'>
        <img src={image} alt={name} className='coinImage' />
      </td>
      <td>{name}</td>
      <td>{symbol.toUpperCase()}</td>
      <td className='coinPrice'>{formatPrice(price)}</td>
      <td>
        {formatPrice(highestPrice)} / {formatPrice(lowestPrice)}
      </td>
      <td className={priceChange < 0 ? 'priceDown' : 'priceUp'}>
        {priceChange.toFixed(2)} %
      </td>
      <td style={{ textAlign: 'right' }}>{volume}</td>
      <td>
        <Link to={`/coins/${id}`}>
          <img src={MoreIcon} alt={name} />
        </Link>
      </td>
    </tr>
  );
};

export default CryptoItem;
