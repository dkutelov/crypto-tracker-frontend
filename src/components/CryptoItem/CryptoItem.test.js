import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

import CryptoItem from './CryptoItem';

describe('Crypto Item Component', () => {
  const item = {
    id: 1,
    name: 'bitcoin',
    image: 'http://www.test.com/some-image.jpg',
    symbol: 'bitcoin',
    current_price: 123,
    high_24h: 125,
    low_24h: 121,
    price_change_percentage_24h: 5.25,
    total_volume: 123123,
  };

  it('Coin image should have proper src', () => {
    render(
      <BrowserRouter>
        <CryptoItem {...item} />
      </BrowserRouter>
    );

    expect(document.querySelector('img').src).toBe(
      'http://www.test.com/some-image.jpg'
    );
  });

  it('Should match snapshot', () => {
    const component = renderer.create(
      <BrowserRouter>
        <CryptoItem {...item} />
      </BrowserRouter>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
