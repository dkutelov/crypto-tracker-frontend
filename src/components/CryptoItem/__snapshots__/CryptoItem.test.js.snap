// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Crypto Item Component Should match snapshot 1`] = `
<tr>
  <td
    className="coinImageWrapper"
  >
    <img
      alt="bitcoin"
      className="coinImage"
      src="http://www.test.com/some-image.jpg"
    />
  </td>
  <td>
    bitcoin
  </td>
  <td>
    BITCOIN
  </td>
  <td
    className="coinPrice"
  >
    123
  </td>
  <td>
    125
     / 
    121
  </td>
  <td
    className="priceUp"
  >
    5.25
     %
  </td>
  <td
    style={
      Object {
        "textAlign": "right",
      }
    }
  >
    123123
  </td>
  <td>
    <a
      href="/coins/1"
      onClick={[Function]}
    >
      <img
        alt="bitcoin"
        src="more.svg"
      />
    </a>
  </td>
</tr>
`;
